src/transformers/modeling_utils.py

```diff
@@ -2641,7 +2642,9 @@ def retrieve_modules_from_names(self, names, add_prefix=False, remove_prefix=Fal
 
         # torch.nn.ParameterList is a special case where two parameter keywords
         # are appended to the module name, *e.g.* bert.special_embeddings.0
-        module_keys = module_keys.union(set([".".join(key.split(".")[:-2]) for key in names if key[-1].isdigit()]))
+        module_keys = module_keys.union(
+            set([".".join(key.split(".")[:-2]) for key in names if key and key[-1].isdigit()])
+        )
```

Comment:
Why the change here? Can we revert if it's just styling?

Reply:
It's not just styling I'm adding `if key and key[-1]` vs `if key[-1]` it should prevent failing even if there is a bad prefix.

I still could revert though. I left it to prevent further failure, but we're already in a bad state if this fails so...

Reply:
It was `""` so `len(key) > 0` it is.